src/components/internal/CompoundField.tsx

```diff
@@ -0,0 +1,66 @@
+/** Internal component to help create compound fields */
+import { useState } from "react";
+import { Css } from "src/Css";
+import { TextFieldInternalProps } from "src/interfaces";
+import { maybeCall } from "src/utils";
+
+export function CompoundField({ children }: { children: JSX.Element[] }) {
+  if (children?.length !== 2) {
+    throw global.Error("CompoundField requires two children components");
+  }
+  const [focusedEl, setFocusedEl] = useState<"left" | "right" | undefined>();
+  const commonStyles = Css.df.aic.fs1.maxwPx(550).bt.bb.bGray300.$;
+  const internalProps: TextFieldInternalProps = { compound: true };
+
+  return (
+    <div css={Css.df.$}>
+      <div
+        css={{
+          ...commonStyles,
+          ...Css.bl.borderRadius("4px 0 0 4px").if(focusedEl === "left").bLightBlue700.$,
+        }}
+      >
+        {{
```

Comment:
I don't think I've seen this syntax...is this like a `cloneElement`?  Seems neat either way.

Reply:
Yea, hopefully this is cool to do. I assume it is like `cloneElement`, but not sure which is really better. I saw what was returned in the `children` prop, and saw I could just clone/expand the props and not the whole element.

Reply:
I switched over to using `cloneElement` after looking at the source code. We can accomplish the same thing, but appears using `cloneElement` would be safer. There are a couple of properties (`child._source` and `child._self`) that don't get copied over when just spreading. They are `undefined` in storybook, but I'm no sure how/when they're populated.